version: "3.3"

volumes:
  tosca-db-data:
  tosca-geoserver-data-dir:

services:
  db:
    image: kartoza/postgis:${POSTGIS_VERSION_TAG}
    volumes:
      - tosca-db-data:/var/lib/postgresql
    ports:
      - ${POSTGRES_PORT}:5432
    environment:
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASS=${POSTGRES_PASS}
      - ALLOW_IP_RANGE=${ALLOW_IP_RANGE}
      - FORCE_SSL=TRUE
    restart: on-failure
    healthcheck:
      test: "PGPASSWORD=${POSTGRES_PASS} pg_isready -h 127.0.0.1 -U ${POSTGRES_USER} -d ${POSTGRES_DB}"

  geoserver:
    image: kartoza/geoserver:${GS_VERSION}
    volumes:
      - tosca-geoserver-data-dir:/opt/geoserver/data_dir
      - ./build_data/:/settings
      # sample volume attachment. Please give full path.
      #- /home/tosca/sample_data:/mnt/data
    ports:
      - ${GEOSERVER_PORT}:8080
    restart: on-failure
    env_file:
      - .env
    depends_on:
      - db
    healthcheck:
      test: "curl --fail --silent --write-out 'HTTP CODE : %{http_code}\n' --output /dev/null -u ${GEOSERVER_ADMIN_USER}:'${GEOSERVER_ADMIN_PASSWORD}' http://localhost:8080/geoserver/rest/about/version.xml"
      interval: 1m30s
      timeout: 10s
      retries: 3
